docs/source/reference/distributions.rst

```diff
@@ -1,31 +1,16 @@
-.. module:: optuna.distributions
-
 optuna.distributions
 ====================
 
-.. autoclass:: UniformDistribution
-    :members:
-
-.. autoclass:: LogUniformDistribution
-    :members:
-
-.. autoclass:: DiscreteUniformDistribution
-    :members:
-
-.. autoclass:: IntUniformDistribution
-    :members:
-    :exclude-members: to_external_repr, to_internal_repr
```

Comment:
The previous docs exclude some methods from some classes, but the new docs don't do so. I would like to exclude the methods which are excluded in the previous docs. Can we realize this point?

Reply:
If I specified it [here](https://github.com/optuna/optuna/pull/1400/files#diff-1ec61223727a10e5e4c74c68336809fcR171), we can do it. 

btw, why are they excluded? If they are for internal usage, we can make them private 🤔

Reply:
Hmm.. I'm not sure about `distributions`, but for `samplers`, they are excluded because they are already explained in `BaseSampler`.

Reply:
Thank you for the comment!
If we do not have strong reasons to exclude documentation, how about showing every public methods in the documentation?
If there are any methods which will not be used by users, we can make them private later.

Reply:
FYI: They will not be excluded after this PR.
```
# IntUniformDistribution, IntLogUniformDistribution, CategoricalDistribution
- to_external_repr
- to_internal_repr
# PyCmaSampler, CmaEsSampler, SkoptSampler
- infer_relative_search_space
- sample_relative
- sample_independent
# FastAIPruningCallback
- on_epoch_end
# optuna.integration.lightgbm.LightGBMTuner
- sample_train_set
# optuna.pruners
- prune
# Study
- system_attrs
- set_system_attr
- storage
```

Reply:
In my understanding, the above methods can be categorized as follows:

### Should be private.
I will make them private in other PR.
- to_external_repr
- to_internal_repr

### They are documented in base class and excluded due to duplication.
Now, each documentation has a different page, so we do not need to exclude them.
- infer_relative_search_space
- sample_relative
- sample_independent
- prune

### Callback method
If you agree the idea that every public API should be documented, we do not need to exclude them.
- on_epoch_end

### I can not judge them
- system_attrs
- set_system_attr
- trial_id
- storage
- sample_train_set

Reply:
Thanks for your all efforts! I would appreciate your careful considerations. Your suggestions are basically LGTM. 

After discussions with @toshihikoyanase and @hvy, I have noticed that `to_external_repr` and `to_internal_repr` can be used by users for **user-defined samplers** (in `suggest_int`, `suggest_discrete`, and `suggest_categorical`). But, the [current tutorial for user-defined samplers](https://optuna.readthedocs.io/en/stable/tutorial/sampler.html) does not include any use-cases of `to_external_repr` and `to_internal_repr` because this tutorial only provides an example to implement `suggest_float`. Therefore, I think
- we don't have to make `to_external_repr` and `to_internal_repr` private, and
- we should add some examples of how to use `to_external_repr` and `to_internal_repr` for user-defined samplers.

I'm so sorry for misunderstanding use-cases of `to_external_repr` and `to_internal_repr`.

Secondly, for the `I can not judge them` section, I think
- `trial_id` is deprecated and removed in https://github.com/optuna/optuna/pull/1371, so we don't have to consider it :), and
- other methods should be documented in public as same as other methods/classes.